### ISSUE TITLE
Better Callback Documentation

### ISSUE THREAD

**Issue by Syntax-Sculptor** [TeamREPENTOGON/REPENTOGON#435]
While the callbacks page is a very useful resource for referencing what REPENTOGON has to offer, it has two faults:

- When a callback's return type is a table, it is not stated most of the time whether it's an array or a dictionary.
- Typos such as incorrect return types and missing parameters.
- Could benefit from examples to onboard beginners. 

- [ ] Provide more examples
- [ ] Fix return types and parameters

**Comment by jsgnextortex** [comment]
I dont know if its appropiate to use the concept of a "dictionary" on lua where they dont really exist, dictionaries are just tables.

**Comment by Syntax-Sculptor** [comment]
> I dont know if its appropiate to use the concept of a "dictionary" on lua where they dont really exist, dictionaries are just tables.

Yeah that's the tricky part right there. There's only a table data type but the way how it's returned should be more clear. Perhaps instead of specifying it as an "array" or "dictionary", maybe a code example for the callback could just speak for itself?

**Comment by jsgnextortex** [comment]
We do those on the docs, the idea is that if someone needs extra information on callbacks they would go check there.

**Comment by Sylmir** [comment]
Indexed array vs. associative array ? Either way, examples should be there
